packages/browser/src/tracing/previousTrace.ts

```diff
@@ -0,0 +1,98 @@
+import type { Span } from '@sentry/core';
+import { logger, SEMANTIC_LINK_ATTRIBUTE_LINK_TYPE, spanToJSON, type SpanContextData } from '@sentry/core';
+import { WINDOW } from '../exports';
+import { DEBUG_BUILD } from '../debug-build';
+
+export interface PreviousTraceInfo {
+  /**
+   * Span context of the previous trace's local root span
+   */
+  spanContext: SpanContextData;
+
+  /**
+   * Timestamp in seconds when the previous trace was started
+   */
+  startTimestamp: number;
+}
+
+// 1h in seconds
+export const PREVIOUS_TRACE_MAX_DURATION = 216_000;
+
+// session storage key
+export const PREVIOUS_TRACE_KEY = 'sentry_previous_trace';
+
+/**
+ * Adds a previous_trace span link to the passed span if the passed
+ * previousTraceInfo is still valid.
+ *
+ * @returns the updated previous trace info (based on the current span/trace) to
+ * be used on the next call
+ */
+export function addPreviousTraceSpanLink(
+  previousTraceInfo: PreviousTraceInfo | undefined,
+  span: Span,
+): PreviousTraceInfo {
+  const spanJson = spanToJSON(span);
+
+  if (!previousTraceInfo) {
+    return {
+      spanContext: span.spanContext(),
+      startTimestamp: spanJson.start_timestamp,
+    };
+  }
+
+  if (previousTraceInfo.spanContext.traceId === spanJson.trace_id) {
+    // This means, we're still in the same trace so let's not update the previous trace info
+    // or add a link to the current span.
+    // Once we move away from the long-lived, route-based trace model, we can remove this cases
+    return previousTraceInfo;
+  }
+
+  if (Date.now() / 1000 - previousTraceInfo.startTimestamp <= PREVIOUS_TRACE_MAX_DURATION) {
```

Comment:
maybe add a short comment here which timestamp this represents :)

Reply:
[be5b526](https://github.com/getsentry/sentry-javascript/pull/15569/commits/be5b5261bd3c3352b88e8721218e0876b28bda40)